### ISSUE TITLE
Upper bounds checking for EVs

### ISSUE THREAD

**Issue by vinnydiehl** [vinnydiehl/pokelog#1]
As it is right now, though the EV inputs and server validate 0-255, the JavaScript on the kill buttons will happily keep updating the inputs client-side. Patching this is the easy part.

The total EVs that a Pokémon can have in all of their stats is 510. When it comes to manually editing the inputs, I will turn the borders all red if this total is exceeded, and if they are maxed out I will turn them green. Also easy.

However, implementing this with the kill buttons has led me to the top of a rabbit hole into which I am about to dive: if you are at 509 EVs, and you receive an EV yield that affects multiple stats, which stat gets it? Are some stats prioritized over others? Does it make a difference if the stats are coming from an item, or a Pokémon with a dual yield? I'm going to have to compose some test cases, and use a save editor to verify, for several generations of games... Starting with Emerald and we'll go from there.

**Comment by vinnydiehl** [comment]
For my first round of testing I created a party of Zigzagoon of varying natures and IVs, with EVs totaling 510 but with room to grow in every stat (distribution randomized), and battled Delcatty, which yield 1 HP and 1 Speed. In every case, HP took the cake. It's looking like EVs are assigned in the order typically listed by the games (HP -> Atk -> Def -> Sp.A -> Sp.D -> Spe), I'll do another round of testing to confirm.

**Comment by vinnydiehl** [comment]
A second round of testing confirms this behavior for Emerald; I made another team of all different Pokémon; different natures, genders, IVs, some legendary, and battled Meganium (1 Def 2 Sp.D, to see if having a higher number of Sp.D would help), and all received the Defense EV.

**Comment by vinnydiehl** [comment]
I ran an experiment in Pokémon Sword in order to see if this behavior is consistent in the modern games, and to see if the new power items make any difference, and it wouldn't appear so; I created another random team, and gave half of them power weights (boosts HP in addition to whatever you're battling) and the other half power anklets (boosts speed similarly). Because of the EXP sharing mechanics in this generation, I only needed to battle one Pokémon that yielded somewhere in between these stats (HP is first, and Speed is last) in order to get some sort of boost, either from the battle or from the item, on each of them. Those with the power weights all received HP boosts, while those with the power anklets all received Attack boosts from the Stufful I battled.
